In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [4]:
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


<h1>1) Data Description</h1>

This project utilizes two datasets provided as CSV files: players.csv and sessions.csv. The data was collected from a Minecraft research server run by a Computer Science group at UBC, where <code>players.csv</code> contains information on individual players, while <code>sessions.csv</code> contains information on individual play sessions.

<h2>Dataset Summaries</h2>
<h3><code>players.csv</code></h3>
<ul>
    <li>196 observations, each representing a unique player</li>
    <li>7 columns/variables</li>
    <li>Variable Details:
    <ol>
        <li>experience (character): A categorical variable describing the player's experience (e.g., 'Pro', 'Veteran', 'Amateur', 'Regular').</li>
        <li>subscribe (boolean/logical vector): A boolean variable (True/False) indicating if the player is subscribed.</li>
        <li>hashedEmail (character): A categorical variable serving as a unique identifier for each player.</li>
        <li>played_hours (double): A quantitative variable for the total number of hours played.</li>
        <li>name (character): A categorical variable for the player's in-game name.</li>
        <li>gender (character): A categorical variable for the player's gender.</li>
        <li>Age (double): A quantitative variable for the player's age.</li>
    </ol>
    </li>
</ul>

<h3><code>sessions.csv</code></h3>
<ul>
    <li>1,535 observations, each representing one play session</li>
    <li>5 columns/variables</li>
    <li>Variable Details:
        <ol>
            <li>hashedEmail (character): A categorical variable serving as a unique identifier for each player.</li>
            <li>start_time (character): The start time of each session as a categorical variable, stored in plain text.</li>
            <li>end_time (character): The end time of each session as a categorical variable, stored in plain text.</li>
            <li>original_start_time (double): The start time of each session as a quantitative variable, stored as a Unix timestamp.</li>
            <li>original_end_time (double): The end time of each session as a quantitative variable, stored as a Unix timestamp.</li>
        </ol>
    </li>
</ul>

<h2>Issues within the Data</h2>

Upon initial analysis of the data, there are a few potential errors that will need to be addressed. <ul> <li>
<code>players.csv</code>'s <code>Age</code> column contains 2 missing/NA values, and <code>sessions.csv</code>'s <code>end_time</code> and <code>original_end_time</code> columns each have 2 missing/NA values. Because these errors are fairly minor, we can choose to remove them or perform mean imputation.</li> <li> Additionally, <code>start_time</code> and <code>end_time</code> would likely need to be converted to a numerical form for any time-based analysis, but we are already given their Unix 'original' counterparts. </li> <li> Finally, and most glaringly, some data is highly skewed. <code>played_hours</code>, for example, has a mean of 5.85 yet a median of 0.10, implying that most players have very few hours while a few players have very many hours. This trend continues into the experience variable as well, where there are significantly more Amateurs (63) than Pros (14). </li></ul>

In [35]:
players_hours_stats <- players |>
    select(played_hours) |>
    summarise(
        mean_played_hours = round(mean(played_hours, na.rm = TRUE), 2),
        median_played_hours = median(played_hours, na.rm = TRUE))

players_experience_stats <- players |>
    group_by(experience) |>
    summarize(count = n())

players_hours_stats
players_experience_stats

mean_played_hours median_played_hours
1 5.85              0.1

experience count
1 Amateur    63   
2 Beginner   35   
3 Pro        14   
4 Regular    36   
5 Veteran    48

[1] "variable"